graphistry/umap_utils.py

```diff
@@ -119,21 +160,30 @@ def umap_lazy_init(
         negative_sample_rate=5,
         n_components: int = 2,
         metric: str = "euclidean",
+        has_umap: str = "True",
+        engine: str = "auto", 
     ):
 
         # FIXME remove as set_new_kwargs will always replace?
-        has_umap, _, umap = lazy_umap_import_has_dependancy()
-
-        if has_umap and not self.umap_initialized:
+        engine_resolved = resolve_umap_engine(engine)
+        if engine_resolved == "umap_learn":
+            _, _, umap_engine = lazy_umap_import_has_dependancy()
+        elif engine_resolved =='cuml':
+            _, _, umap_engine = lazy_cuml_import_has_dependancy()
+        else:
+            raise ValueError("No umap engine, ensure 'auto', 'umap', or 'cuml', and the library is installed")
+        
+        if has_umap and not self.umap_initialized: 
             umap_kwargs = dict(
                 n_components=n_components,
-                metric=metric,
+                # metric=metric,
```

Comment:
why remove metric?

if a problem for cuml, we can throw a targeted exn if engine is cuml

Reply:
as far as I can tell this flag is still an open issue, with some hacks to get it working https://github.com/rapidsai/cuml/issues/1653#issuecomment-765652438. From what I can tell there is no such flag listed in the UMAP function https://docs.rapids.ai/api/cuml/stable/api.html?highlight=umap#cuml.UMAP

Reply:
yeah that was my guess why you did it. it ends up big for umap users, so to support mult engines, a pattern can be like:

```python
opts = {
  a=b,
  c=d,
  **({'x': 'y'} if engine='umap' else {})
}
```